apps/settings/js/panels/call/panel.js

```diff
@@ -0,0 +1,472 @@
+        this._elements.voiceMailItem.onclick = () => {
+          DialogService.show('call-voiceMailSettings');
+        };
+      },
+
+      _initCallWaitingAlertLabelClickEvent: function() {
+        this._elements.alertLabel.onclick = () => {
+          this._elements.alertPanelConfirmInput.checked = false;
+          this._elements.alertPanel.hidden = false;
+        };
+      },
+
+      _initVoicePrivacyModeChangeEvent: function() {
+        var defaultVoicePrivacySettings = [].map.call(this._conns, () => {
+          return false;
+        });
+
+        var voicePrivacyHelper = SettingsHelper(
+          'ril.voicePrivacy.enabled', defaultVoicePrivacySettings);
+
+        this._elements.voicePrivacyInput.onchange = () => {
+          var checked = this._elements.voicePrivacyInput.checked;
+          voicePrivacyHelper.get((values) => {
+            var originalValue = !checked;
+            this._conn.setVoicePrivacyMode(checked).then(() => {
+              values[this._cardIndex] = !originalValue;
+              voicePrivacyHelper.set(values);
+            }, () => {
+              // restore the value if failed.
+              this._elements.voicePrivacyInput.checked = originalValue;
+            });
+          });
+        };
+      },
+
+      _initCallWaitingAlertCancelButtonClickEvent: function() {
+        this._elements.alertPanelCancelButton.onclick = () => {
+          this._elements.alertPanel.hidden = true;
+        };
+      },
+
+      _addVoiceTypeChangeListeners: function() {
+        var voiceTypes = [].map.call(this._conns, () => {
+          return null;
+        });
+
+        [].forEach.call(this._conns, (conn, index) => {
+          voiceTypes[index] = conn.voice.type;
+          conn.addEventListener('voicechange', () => {
+            var newType = conn.voice.type;
+            if (index !== this._cardIndex || voiceTypes[index] === newType) {
+              return;
+            }
+
+            voiceTypes[index] = newType;
+
+            this._updateNetworkTypeLimitedItemsVisibility(newType);
+            this._refreshCallSettingItems();
+          });
+        });
+      },
+
+      _updateNetworkTypeLimitedItemsVisibility: function(voiceType) {
+        // The following features are limited to GSM types.
+        this._elements.callForwardingItem.hidden =
+        this._elements.callBarringItem.hidden =
+        this._elements.callWaitingItem.hidden =
+        this._elements.callerIdItem.hidden =
+          (CallConstant.NETWORK_TYPE_CATEGORY[voiceType] !== 'gsm');
+
+        // The following feature is limited to CDMA types.
+        this._elements.voicePrivacyItem.hidden =
+          (CallConstant.NETWORK_TYPE_CATEGORY[voiceType] !== 'cdma');
+      },
+
+      _refreshCallSettingItems: function() {
+        this._updateVoiceMailItemState();
+        this._updateFdnStatus();
+        this._updateVoicePrivacyItemState();
+        this._updateCallerIdPreference();
+        this._updateCallerIdItemState();
```

Comment:
I'm not sure why it look like this in the original implementation. However, `_updateCallerIdItemState` is supposed to be called after `_updateCallerIdPreference` resolved.